### PR TITLE
Fail incomplete downloads

### PR DESCRIPTION
**Pull request by cavis**
Same changes as PRX/cms-audio-lambda#6

### CODE REVIEW

**Comment by cavis** [comment]
Reminder: image-lambda tests not currently working in Snap, due to the native dependencies for `sharp`.  Passes locally.  I'm willing to live with that, for the remainder of Snap's lifetime.

File: lib/resizer.js [hunk 1]
```diff
@@ -134,27 +145,20 @@ exports.download = (uploadUrl) => {
...
-        if (resp.statusCode === 200) {
-          tmpStream.on('finish', () => resolve(result));
-          req.pipe(tmpStream);
-          req.resume();
-        } else {
+      (uploadUrl.match(/^https/) ? https : http).get(uploadUrl, resp => {
+        if (resp.statusCode !== 200) {
           reject(new Error(`Got ${resp.statusCode} for url: ${uploadUrl}`));
+        } else {
+          result.contentType = resp.headers['content-type'];
```
Comment (kookster) [inline lib/resizer.js:155]: Doesn't seem to be used for anything, but doesn't hurt either

**Review by kookster — APPROVED** [review]
lgtm

**Comment by kookster** [comment]
+1 I also like the setError addition for being able to see what happened

**Pull request merged**